packages/core/src/shader/utils/generateUniformBufferSync.ts

```diff
@@ -0,0 +1,295 @@
+import type { Dict } from '@pixi/utils';
+import { mapSize } from '../utils';
+import { IUniformData } from '../Program';
+import { UniformGroup } from '../UniformGroup';
+import { uniformParsers } from './uniformParsers';
+import type { Renderer } from '../../Renderer';
+import type { Buffer } from '../../geometry/Buffer';
+
+export type UniformsSyncCallback = (...args: any[]) => void;
+
+const uboUpdate =  (_ud: any, _uv: any, _renderer: Renderer, _syncData: any, buffer: Buffer) =>
+{
+    _renderer.buffer.update(buffer);
+};
+
+// cv = CachedValue
+// v = value
+// ud = uniformData
+// uv = uniformValue
+// l = location
+const UBO_TO_SINGLE_SETTERS: Dict<string> = {
+    float: `
+        data[offset] = v;
+    `,
+    vec2: `
+        data[offset] = v[0];
+        data[offset+1] = v[1];
+    `,
+    vec3: `
+        data[offset] = v[0];
+        data[offset+1] = v[1];
+        data[offset+2] = v[2];
+
+    `,
+    vec4: `
+        data[offset] = v[0];
+        data[offset+1] = v[1];
+        data[offset+2] = v[2];
+        data[offset+3] = v[3];
+    `,
+    mat2: `
+        data[offset] = v[0];
+        data[offset+1] = v[1];
+
```

Comment:
I guess we're assuming that the third column of the matrix is always [0, 0, 1]. Same for mat3. It would be nice to make this explicit. But that's not a biggie.

Reply:
in this instance it should set all values directly from the array for both mat2 and mat3?